src/storage/mapping/BaseFileIdentifierMapper.ts

```diff
@@ -41,12 +40,6 @@ export class BaseFileIdentifierMapper implements FileIdentifierMapper {
    */
   public async mapUrlToFilePath(identifier: ResourceIdentifier, isMetadata: boolean, contentType?: string):
```

Comment:
Does this means that FixedContentTypeMapper, ExtensionBasedMapper, FileIdentifierMapper and all files + tests depending on those will also have to be changed?

Reply:
Not `FileIdentifierMapper` since there should be no changes to the interface, but `FixedContentTypeMapper` and `ExtensionBasedMapper` will have to change in that they will have to take the metadata strategy as input. Besides that I don't expect too many changes would be necessary. Just in the places where it's checked if the identifier targets a meta resource.

Reply:
So I had an deeper look into the code and I think a change might not be needed any more. `isMetadataPath` is a valid function, if it is being called on a file path. The issue occurs when it is being called on an identifier, which happens for example here: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/b3da9c9fcfb642e8a8a6d537b2e69cbd74ee8a88/src/storage/mapping/BaseFileIdentifierMapper.ts#L47-L49

But that block happens to also be removed in this PR so is actually not relevant any more. And having a look at the other usages, they all seem to be used on file paths. This is fine, since the `BaseFileIdentifierMapper` is hardcoded to attach `.meta` to file paths to indicate metadata: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/b3da9c9fcfb642e8a8a6d537b2e69cbd74ee8a88/src/storage/mapping/BaseFileIdentifierMapper.ts#L51-L53
So perhaps no change is needed. So I would suggest just doing some small tests locally where you change the metadata strategy configuration to use a different extension (`.atem` for example) and see if everything still works correctly when using a file-based backend. If yes, no change is probably needed here.

Reply:
My postman tests seemed to work fine when using the `.atem` as suffix for metadata.